File: retrocast-client/Retrocast/Sources/Models/Channel.swift
import Foundation

enum ChannelType: Int, Codable, Sendable {
    case text = 0
    case voice = 2
    case category = 4
}

struct Channel: Codable, Identifiable, Hashable, Sendable {
    let id: Snowflake
    let guildID: Snowflake
    var name: String
    let type: ChannelType
    var position: Int
    var topic: String?
    var parentID: Snowflake?

    enum CodingKeys: String, CodingKey {
        case id
        case guildID = "guild_id"
        case name
        case type
        case position
        case topic
        case parentID = "parent_id"
    }
}

File: retrocast/internal/models/channel.go
package models

type ChannelType int

const (
	ChannelTypeText     ChannelType = 0
	ChannelTypeVoice    ChannelType = 2
	ChannelTypeCategory ChannelType = 4
)

type Channel struct {
	ID       int64       `json:"id,string"`
	GuildID  int64       `json:"guild_id,string"`
	Name     string      `json:"name"`
	Type     ChannelType `json:"type"`
	Position int         `json:"position"`
	Topic    *string     `json:"topic,omitempty"`
	ParentID *int64      `json:"parent_id,string,omitempty"`
}
